nmdc_runtime/api/endpoints/users.py

```diff
@@ -32,6 +35,43 @@
 router = APIRouter()
 
 
+# @router.get("/orcid_cookie_test", include_in_schema=False)
+# def orcid_cookie_test():
+#     return HTMLResponse(
+#         '<img src="/static/ORCIDiD_icon128x128.png" />'
+#         f'<a href="https://orcid.org/oauth/authorize?client_id={ORCID_CLIENT_ID}'
+#         "&response_type=code&scope=openid&"
+#         f'redirect_uri={BASE_URL_EXTERNAL}/orcid_code">Login with ORCiD</a>'
+#     )
+
+
+# {{GLOBALS_orcid_authorize_url}}&state={{request.url.path}}
+
+
+@router.get("/orcid_code", response_class=RedirectResponse)
+async def receive_orcid_code(request: Request, code: str, state: str | None = None):
+    rv = requests.post(
+        "https://orcid.org/oauth/token",
+        data=(
+            f"client_id={ORCID_CLIENT_ID}&client_secret={ORCID_CLIENT_SECRET}&"
+            f"grant_type=authorization_code&code={code}&redirect_uri={BASE_URL_EXTERNAL}/orcid_code"
+        ),
+        headers={
+            "Content-type": "application/x-www-form-urlencoded",
+            "Accept": "application/json",
+        },
+    )
+    token_response = rv.json()
+    response = RedirectResponse(state or request.url_for("custom_swagger_ui_html"))
+    for key in ["user_orcid", "user_name", "user_id_token"]:
+        response.set_cookie(
+            key=key,
+            value=token_response[key.replace("user_", "")],
+            max_age=2592000,
+        )
+    return response
+
+
 @router.get("/orcid_authorize")
 async def orcid_authorize():
```

Comment:
@dwinston Can we get rid of /orcid_authorize?

Reply:
Definitely